protos/sigstore_common.proto

```diff
@@ -36,6 +38,33 @@ enum HashAlgorithm {
         SHA2_512 = 2;
 }
 
+// Subset of known signature algorithms.
+enum SignatureAlgorithm {
+        SIGNATURE_ALGORITHM_UNSPECIFIED = 0;
+        ECDSA_P256_SHA_256 = 1; // See NIST FIPS 186-4
+        ECDSA_P256_HMAC_SHA_256 = 2; // See RFC6979
+        ED25519 = 3; // See RFC8032
+        RSA_PKCS1V5 = 4; // See RFC8017
+        RSA_PSS = 5; // See RFC8017
+}
+
+// The permutations of signature algorithms and encoding schemas.
+// To avoid contradicting formats such as PKCS1 with ED25519 the valid
+// permutations are listed as a linear set instead of a cartesian set.
+enum KeyFormat {
```

Comment:
Nit: maybe "PublicKeyFormat"?

Reply:
Also, I wonder if we could make it clearer that SignatureAlgorithm is a fixed function of KeyFormat. Not sure how to do that

Reply:
How about:
* `PublicKeyDescription`
* `PublicKeyDetails`
* `PublicKeyParameters`
* `PublicKeyAttributes`
* `PublicKeyMetaData`

I think I would go with `PublicKeyDetails`. The name is vague, but with good comments we can solve it.